### ISSUE TITLE
Feature request: Switch from Sass imports to Sass Module System (`@import` -> `@use`)

### ISSUE THREAD

**Issue by simPod** [FortAwesome/Font-Awesome#19167]
### Is your feature request related to a problem?

Sass `@import` is deprecated. See https://sass-lang.com/documentation/at-rules/import

### Feature description

Current fontawesome sass is built around `@import` and should be migrated to `@use`.

### Feature request checklist

- [X] The title starts with "Feature request: " and is followed by the requested feature description
- [X] This is a single feature
- [X] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome/issues) and to the best of my knowledge this is not a duplicate

**Comment by tagliala** [comment]
Hi!

Thanks for this heads-up.

From [Sass docs](https://sass-lang.com/documentation/at-rules/import):

> Prefer the [@use rule](https://sass-lang.com/documentation/at-rules/use) instead. (Note that only Dart Sass currently supports @use. Users of other implementations must use the @import rule instead.)

Font Awesome supports a variety of Sass compilers, including old ones and [scssphp](https://scssphp.github.io/scssphp/), so just switching from `@import` to `@use` would be a breaking change.

We were able to provide a backward-compatible version of the division deprecation #17946, but I don't know if there is the chance to do the same with this one

Is this already printing deprecation warnings to console in some dart-sass versions?

@robmadole any thoughts? I think this should be postponed as late as possible

**Comment by robmadole** [comment]
@tagliala I agree on postponing. We want to support as many compilers as possible. I don't think we'd introduce `@use` until FA version 7. The way I'm thinking about it right now is it requires a major version change. I guess we could start providing legacy and modern versions of our .scss files. But that complicates things for what return?

**Comment by simPod** [comment]
I did not actually expect this to land in v6 but wanted to raise issue so it's on a roadmap. It's obviously a BC, e.g. node-sass is already deprecated but since v6 supports it, it cannot be broken now.

> Font Awesome supports a variety of Sass compilers, including old ones and [scssphp](https://scssphp.github.io/scssphp/),

I also think this should be reconsidered in the future and only those maintained ones should be supported.

**Comment by rocket-turtle** [comment]
Are there any plans when v7 will be developed?
